src/MessageEditor.js

```diff
@@ -14,23 +21,69 @@ class MessageEditor extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (this.props.id !== prevProps.id && this.props.id) {
+      this.setState({
+        messages: [],
+        gauges: []
+      });
       api.fetch(`glyphs/${this.props.id}/messages`)
       .then(messages => this.setState({ messages: messages }));
 
       api.fetch(`glyphs/${this.props.id}/gauges`)
       .then(gauges => this.setState({ gauges: gauges }));
+
     }
   }
 
   render() {
-    const { messages, gauges } = this.state;
+    let { messages, gauges } = this.state;
+
+    const addVMessage = () => {
+      api.post(`glyphs/${this.props.id}/messages/`, {"pass" : this.inputRef.value, "text" : 'Default message', "probability" : 0}).then(result => {console.log(result)})
+      messages.push({"text" : 'Default message', "probability" : 0})
+      this.forceUpdate();
+    }
+
+    const password = () => {
+      this.forceUpdate();
```

Comment:
Why exactly is this needed?

Reply:
I am asking this mostly because `this.forceUpdate()` should be avoided whenever possible.  The "React" way of doing this I believe would be to have the value of the password stored in the component state and then setting `value` of `TextField` in the component.

https://reactjs.org/docs/forms.html#controlled-components

Reply:
That way, editing the field edits the state, and automatically updates.